lib/backend/k8s/conversion.go

```diff
@@ -224,18 +224,16 @@ func (c converter) podToWorkloadEndpoint(pod *kapiv1.Pod) (*model.KVPair, error)
 	profile := fmt.Sprintf("ns.projectcalico.org/%s", pod.ObjectMeta.Namespace)
 	workload := fmt.Sprintf("%s.%s", pod.ObjectMeta.Namespace, pod.ObjectMeta.Name)
 
-	// If the pod isn't ready, we can't parse it.
```

Comment:
I don't think so - this is just a conversion method, and I don't think it should make assumptions about whether or not the calling code is doing the right thing.  The calling code should know what conditions it needs to check, but this bit of code should just do its best to turn a Pod into a WorkloadEndpoint.